src/session.js

```diff
@@ -171,17 +171,18 @@ class Session {
     if (this.suspendResume.isSuspended) {
       return this.Promise.reject(new Error('Session suspended'));
     }
-    const data = Object.assign({}, this.protocol, {
-      method: request.method,
-      handle: request.handle,
-      params: request.params,
-      delta: request.delta,
-    });
-    const response = this.rpc.send(data);
-    request.id = data.id;
-    request.retry = () => this.send(request);
-
-    const promise = this.intercept.execute(this, response, request);
+    request.id = this.rpc.createRequestId();
+    const promise = this.intercept.executeRequests(this, this.Promise.resolve(request))
```

Comment:
Technically, interceptors are just function part of one call chain with a `rpc.send()` in the middle. We may want to rethink this for v3 to streamline it into one call chain instead, allowing an advanced user to customize it however they want.

Reply:
Yes, I agree, `Session.send(...)` is becoming increasingly complex.